Classes/Package/CustomPages.php

```diff
@@ -164,49 +162,44 @@ public function optionsPage() {
 		}
 	}
 
-	public function footerJs() {
+	public function pageState( $post_states, $post ) {
 
 		if ( function_exists( 'get_field' ) ) {
-			$screen = get_current_screen();
-
-			if ( 'page' == $screen->id ) {
-				$page_pt = false;
-				foreach ( $this->getPosttypeItems() as $pt => $name ) {
-					$field = get_field( "page_for_$pt", 'options' );
-					if ( $field && $field == $_GET['post'] ) {
-						$page_pt = $name;
+			$items = array_merge( $this->getPosttypeItems(), $this->getSpecialItems() );
+
+			foreach ( $items as $key => $name ) {
+				$page_id = get_field( "page_for_$key", 'options' );
+
+				if ( intval( $page_id ) == $post->ID ) {
+					if ( get_post_type_object( $key ) ) {
+						$post_states[] = sprintf( __( 'Archive Page for "%s"', 'sha' ), $name );
+					} else {
+						$post_states[] = sprintf( __( 'Page for "%s"', 'sha' ), $name );
 					}
 				}
+			}
+		}
 
-				if ( ! $page_pt ) {
-					return;
-				}
+		return $post_states;
+	}
 
-				// translators: Du bearbeitest gerade die Seite, die als Übersicht über "Posts" definiert wurde. Der Permalink wird deshalb automatisch überschrieben und die Inhalte können je nach Verwendung im Theme abweichen.
-				$infotext = sprintf( __( 'Du bearbeitest gerade die Seite, die als Übersicht über "%s" definiert wurde. Der Permalink wird deshalb automatisch überschrieben und die Inhalte können je nach Verwendung im Theme abweichen.', 'sha' ), "<b>{$page_pt}</b>" );
-
-				echo '<script id="sayhello_CustomPage">';
-				echo 'jQuery(function($){';
-				echo '$("#titlediv").append("<div class=\"notice notice-warning inline\" style=\"margin-top:20px;\"><p>' . addslashes( $infotext ) . '</p></div>");';
-				echo '});';
-				echo '</script>';
-			} elseif ( 'edit-page' == $screen->id ) {
-				$items = array_merge( $this->getPosttypeItems(), $this->getSpecialItems() );
-
-				echo '<script id="sayhello_CustomPage">';
-				echo "jQuery(function($){\n";
-
-				foreach ( $items as $key => $name ) {
-					$page_id = get_field( "page_for_$key", 'options' );
-					if ( 'page' == get_post_type( $page_id ) ) {
-						// translators: "Posts" Page
-						$text = sprintf( __( '"%s" Page', 'sha' ), $name );
-						echo "$('tbody#the-list tr#post-$page_id td.title strong').append(' — <span class=\"post-state\">$text</ span>');\n";
+	public function editPageNotification() {
+		if ( function_exists( 'get_field' ) ) {
+			$items = array_merge( $this->getPosttypeItems(), $this->getSpecialItems() );
+
+			foreach ( $items as $key => $name ) {
+				$page_id = get_field( "page_for_$key", 'options' );
+
+				if ( intval( $page_id ) == get_the_ID() ) {
+					if ( get_post_type_object( $key ) ) {
+						$class    = 'notice notice-warning';
+						$infotext = sprintf(
+							__( 'Du bearbeitest gerade die Seite, die als Übersicht über "%s" definiert wurde. Der Permalink wird deshalb automatisch überschrieben und die Inhalte können je nach Verwendung im Theme abweichen.', 'sha' ),
```

Comment:
Bitte die richtige Anführungszeichen immer anwenden.

```suggestion
							__( 'Du bearbeitest gerade die Seite, die als Übersicht über «%s» definiert wurde. Der Permalink wird deshalb automatisch überschrieben und die Inhalte können je nach Verwendung im Theme abweichen.', 'sha' ),
```